File: CSA_HW2/Beast.cpp
#include "Beast.h"

//------------------------------------------------------------------------------
// ------------------- CPP FILE OF CHILD CLASS 'FISH' --------------------------
//------------------------------------------------------------------------------

// Implementation of the method for printing information about beast.
void Beast::Output(std::ofstream &output_file_stream) {
    output_file_stream << "Animal " << name_ << " with weight " << weight_ << " grams. \n";
    output_file_stream << "It's quotient from name to weight is " << Quotient() << ". \n";
    output_file_stream << "It's a beast and it's ";
    switch (nutrition_) {
        case 1: output_file_stream << "carnivores.\n";
            break;
        case 2: output_file_stream << "herbivores.\n";
            break;
        case 3: output_file_stream << "insectivores.\n";
            break;
        default: output_file_stream << "omnivores.\n";
    }
}

File: CSA_HW3/Animals/Beast.py
from Animals.Animal import Animal


# ------------------------------------------------------------------------------
# --------------------------- CHILD CLASS 'BEAST' ------------------------------
# ------------------------------------------------------------------------------


class Beast(Animal):
    nutrition_types = ["carnivore", "herbivore", "insectivore", "omnivore"]

    # Constructor for child class beast with extra parameter - nutrition type.
    def __init__(self, name, weight, nutrition):
        super(Beast, self).__init__(name, weight)
        self.__nutrition__ = nutrition

    # Method for printing information about beast.
    def out(self):
        # Calling method for print basic info about beast.
        info = super(Beast, self).out()
        info += "It's a beast and it's a"
        if self.__nutrition__ < 1 or self.__nutrition__ > 4:
            print("Beast nutrition type was incorrect so it was changed to default value (omnivores)")
            self.__nutrition__ = 4
        if 3 <= self.__nutrition__ <= 4:
            info += "n"
        info += " " + self.nutrition_types[self.__nutrition__ - 1] + ".\n"
        return info
